gui/src/pages/NewProjectConfig.svelte

```diff
@@ -0,0 +1,149 @@
+<script lang="ts">
+  import Layout from '../components/Layout.svelte';
+  import Textbox from '../components/Textbox.svelte';
+  import Spinner from '../components/Spinner.svelte';
+  import ErrorLabel from '../components/ErrorLabel.svelte';
+  import SubmitButton from '../components/form/SubmitButton.svelte';
+  import CenterFormPanel from '../components/form/CenterFormPanel.svelte';
+  import { api, isClientError } from '../lib/api';
+  import * as router from 'svelte-spa-router';
+
+  export let params = { starter: '' };
+
+  const { starter } = params;
+  const isEmptyProject = starter === 'empty'; // XXX Use this with the `createProject` api to avoid a second (template URL) parameter if the project is empty.
+
+  let name = starter;
+  let error: Error | null = null;
+
+  let workingDirectory: string | null = null;
+
+  const setWorkingDirectory = (dir: string) => {
+    workingDirectory = dir;
+  };
+
+  (async () => {
+    setWorkingDirectory(await api.kernel.getUserHomeDir());
+  })();
+</script>
+
+<Layout>
+  <CenterFormPanel title={`New project: ${starter}`} backRoute="#/new-project">
+    <form
+      on:submit|preventDefault={async () => {
+        error = null;
+        let workspaceId;
+        // XXX Replace this old createWorkspace with templated createProject
+        try {
+          workspaceId = await api.kernel.createWorkspace(
+            workingDirectory, // Currently this doesn't actually create the new directory, see XXX note above
+          );
+          router.push(`/workspaces/${encodeURIComponent(workspaceId)}`);
+        } catch (ex) {
+          if (!isClientError(ex)) {
+            throw ex;
+          }
+          error = ex;
+        }
+        // XXX Hack to address lack of GUI for applying procfiles, etc.
```

Comment:
@BenElgar I janked this in there for release a long while ago. I think we should make createProject do an apply automatically.

Reply:
Yeah, it should, and also I just realized that we not only need to set up the project such that it is runnable but we might also want some intial step analogous to `npm i`

Even if the template sets up your project perfectly and adds components to exo, if you don't have dependencies installed the process will fail

Reply:
Maybe that's best left up to the user, but I'm not sure

Reply:
So I actually set out to do that but I couldn't immediately see how to acquire a handle on a `Workspace` from a `Kernel`.  Will figure it out.

Reply:
To be clear, when I say that I set out to do that, I mean call apply. I think the user should have the chance to do `npm i` or whatever else before we start the project for them.

Reply:
Overall I think the best template experience would be if we automated that as well.

Give me the full stack project, all hooked up together, with dependencies installed, and already running in dev mode. That way I can already play around with it and start making changes immediately.

Reply:
That being said, I can see some arguments for not doing that.